.github/workflows/agent.yaml

```diff
@@ -13,6 +13,28 @@ jobs:
         with:
           fetch-depth: 0
       - uses: extractions/setup-just@v2
+
+      # Install R with rig to ensure we are compatible
+      - name: Install rig with apt
+        if: ${{ runner.os == 'Linux' }}
+        run: |
+          sudo curl -L https://rig.r-pkg.org/deb/rig.gpg -o /etc/apt/trusted.gpg.d/rig.gpg
+          sudo sh -c 'echo "deb http://rig.r-pkg.org/deb rig main" > /etc/apt/sources.list.d/rig.list'
+          sudo apt update
+          sudo apt install -y r-rig
+      - name: Install rig with brew
+        if: ${{ runner.os == 'macOS' }}
+        run: |
+          brew tap r-lib/rig
+          brew install --cask rig
+      - name: Install rig with Choco
+        if: ${{ runner.os == 'Windows' }}
+        run: choco install rig
+
+      - name: Install R with rig
+        run: rig add release
+      - name: Install renv package
+        run: Rscript -e "install.packages('renv')"
```

Comment:
We could consider running these functional style tests in a separate action, though IMHO that's a refinement for later.

Reply:
Sounds good, would you mind creating a follow-up issue for that when we merge this?

I found it a bit confusing these were part of the unit tests at first blush, but understand it a bit more after digging in. The separation would make it a bit easier to understand right from the start.

Reply:
> This should be its own GitHub Action on the marketplace :)

There actually is [a set of actions provided by r-lib](https://github.com/r-lib/actions) which is great and does lots of good things *but* it doesn't use rig (and thus doesn't have hit the `R.bat` / `cmd.exe` quoting issues). 🫠 I'll add a comment about this just so it's clear _why_ we are doing this kind of install.